src/Fixer/StringNotation/SingleQuoteFixer.php

```diff
@@ -54,17 +54,36 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
             }
 
             $content = $token->getContent();
+            $firstCharacter = $content[0];
 
             if (
-                '"' === $content[0] &&
-                false === strpos($content, "'") &&
-                // regex: odd number of backslashes, not followed by double quote or dollar
-                !Preg::match('/(?<!\\\\)(?:\\\\{2})*\\\\(?!["$\\\\])/', $content)
+                '"' === $firstCharacter &&
+                $this->compatibleWithSingleQuotes($content)
             ) {
                 $content = substr($content, 1, -1);
                 $content = str_replace(array('\\"', '\\$'), array('"', '$'), $content);
                 $tokens[$index] = new Token(array(T_CONSTANT_ENCAPSED_STRING, '\''.$content.'\''));
             }
+
+            if (
+                0 === stripos($content, 'b"') &&
```

Comment:
would `'b"' === substr($content, 0, 2)` be preferred here?

Reply:
I would you the one from your comment, as there is no need to search the whole content for `b"` which is what `stripos` will do, but just my vote ;)